tests/basetest.h

```diff
@@ -0,0 +1,131 @@
+/*
+  basetest.h
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Filipe Azevedo <filipe.azevedo@kdab.com>
+
+  Licensees holding valid commercial KDAB GammaRay licenses may use this file in
+  accordance with GammaRay Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef BASETEST_H
+#define BASETEST_H
+
+#include <config-gammaray.h>
+
+#include <QObject>
+#include <QSignalSpy>
+#include <QAbstractItemModel>
+
+#include <QtTest/qtest.h>
+
+class BaseTest : public QObject
```

Comment:
Good idea to factor out those utilities. They don't need to be in the inheritance hierarchy, right? So we could have them in a separate utility namespace for example, would make it easier to combine this with the other base classes.